packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -1604,6 +1606,24 @@ class OverlayPortalController {
   /// rebuilt.
   void toggle() => isShowing ? hide() : show();
 
+  final List<VoidCallback> _listeners = <VoidCallback>[];
+
+  void _notifyListeners() {
+    for (final VoidCallback listener in _listeners) {
+      listener();
+    }
+  }
+
+  @override
+  void addListener(VoidCallback listener) {
+    _listeners.add(listener);
+  }
+
+  @override
+  void removeListener(VoidCallback listener) {
+    _listeners.remove(listener);
+  }
+
```

Comment:
Without `dispose` the owner of the controller won't be able to properly release all listeners (as it doesn't know what listeners are currently registered). This changes the semantics of the class (see https://github.com/flutter/flutter/issues/138828#issuecomment-2274820633).